### PR TITLE
[DQ_py.cpp] Added the methods Q4 and Q8.

### PR DESCRIPTION
**Pull request by juanjqo**
@dqrobotics/developers 

Hi @mmmarinho, 

This PR adds the bindings for the methods Q4 and Q8, which were recently added in C++ in [this PR](https://github.com/dqrobotics/cpp/pull/56) by Marcos. I added the tests [here](https://github.com/dqrobotics/python-tests/pull/6). 

Best regards, 

Juancho

-passing-passing)

### CODE REVIEW

File: src/DQ_py.cpp [hunk 1]
```diff
@@ -69,6 +69,8 @@ void init_DQ_py(py::module& m)
     dq.def("sharp"               ,&DQ::sharp,                 "Retrieves the sharp of a DQ");
     dq.def("Ad"                  ,&DQ::Ad,                    "Retrieves the adjoint transformation of a DQ.");
     dq.def("Adsharp"             ,&DQ::Adsharp,               "Retrieves the adjoint sharp transformation of a DQ.");
+    dq.def("Q4"                  ,&DQ::Q4,                    "Retrieves the partial derivative of the unit quaternion with respect to its logarithm.");
+    dq.def("Q8"                  ,&DQ::Q8,                    "Retrieves the partial derivative of a DQ with respect to its logarithm.");
```
Comment (mmmarinho) [inline src/DQ_py.cpp]: a unit DQ?

File: src/DQ_py.cpp [hunk 2]
```diff
@@ -126,6 +128,8 @@ void init_DQ_py(py::module& m)
     m.def("Adsharp"             ,&DQ_robotics::Adsharp,              "Retrieves the adjoint sharp transformation of a DQ.");
     m.def("cross"               ,&DQ_robotics::cross,                "Returns the result of the cross product between two DQ.");
     m.def("dot"                 ,&DQ_robotics::dot,                  "Returns the result of the dot product between two DQ.");
+    m.def("Q4"                  ,&DQ_robotics::Q4,                   "Retrieves the partial derivative of the unit quaternion with respect to its logarithm.");
+    m.def("Q8"                  ,&DQ_robotics::Q8,                   "Retrieves the partial derivative of a DQ with respect to its logarithm.");
```
Comment (mmmarinho) [inline src/DQ_py.cpp]: a unit DQ?

**Review by mmmarinho — CHANGES_REQUESTED** [review]
@juanjqo 
Nothing major.

File: src/DQ_py.cpp (hunk 1 above)
Comment (juanjqo) [reply src/DQ_py.cpp]: @mmmarinho fixed!

File: src/DQ_py.cpp (hunk 2 above)
Comment (juanjqo) [reply src/DQ_py.cpp]: @mmmarinho fixed!

**Review by mmmarinho — APPROVED** [review]

**Pull request merged**